### PR TITLE
fix: s3 bucket ownership issues

### PR DESCRIPTION
**Pull request by cpwc**
Couple of changes made to how S3 bucket handles ACL and ownership. 

Breaking change was made to create logging bucket. I've added a default so that it does not break for users from the older version that do not want to have bucket access logging enabled.

### CODE REVIEW

File: aws_s3_bucket.log_bucket.tf [hunk 1]
```diff
@@ -22,17 +22,12 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "log_bucket" {
...
 
 resource "aws_s3_bucket_logging" "name" {
+  count = var.log_bucket_logging != null ? 1 : 0
+
   bucket = aws_s3_bucket.log_bucket.bucket
 
   target_bucket = var.log_bucket_logging
   target_prefix = "${aws_s3_bucket.log_bucket.bucket}/"
-
-}
```
Comment (JamesWoolfenden) [inline aws_s3_bucket.log_bucket.tf]: did you mean to delete the acl to allow logging?

File: aws_s3_bucket.log_bucket.tf [hunk 2]
```diff
@@ -43,3 +38,18 @@ resource "aws_s3_bucket_versioning" "log_bucket" {
     mfa_delete = var.log_bucket_mfa_delete
   }
 }
+
```
Comment (JamesWoolfenden) [inline aws_s3_bucket.log_bucket.tf]: does this replace the acl?

**Comment by cpwc** [comment]
Have rebased to latest master and fixed conflicts.

**Comment by cpwc** [comment]
Pushed a fix on the resources being moved for clarity.

**Review by JamesWoolfenden — APPROVED** [review]
Thanks for the edits, i haven't used this module for a while hopefully its bang up to date now!

**Pull request merged**